command/agent/cache/lease_cache.go

```diff
@@ -814,3 +853,44 @@ func (c *LeaseCache) RegisterAutoAuthToken(token string) error {
 
 	return nil
 }
+
+type clearInput struct {
+	ClearType string
+
+	RequestPath   string
+	Namespace     string
+	Token         string
+	TokenAccessor string
+	Lease         string
+}
+
+func parseClearInput(req *cacheClearRequest) (*clearInput, error) {
+	if req == nil {
+		return nil, errors.New("nil request options provided")
+	}
+
+	if req.Type == "" {
+		return nil, errors.New("no type provided")
+	}
+
+	in := &clearInput{
+		ClearType: req.Type,
+	}
+
+	if req.Namespace != "" {
```

Comment:
This `if` condition can be avoided and Namespace can be set in the above definition of `in`.